### PR TITLE
fix(ci/nightly): ensure release is published

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Sometimes, nightly release was getting marked as draft. Draft releases are visible only to maintainers. 
Reason for why release was marked draft unknown, won't invest too much effort in 3rd party GitHub Actions.

## Changes proposed in this pull request

Add another step to ensure that the release is published, not marked as draft. This step won't fail if re-run on published release.

### CODE REVIEW

File: .github/workflows/nightly-build.yaml [hunk 1]
```diff
@@ -52,3 +52,8 @@ jobs:
             > [!warning]
             > The Nightly build is for adventurous folks. It's updated daily with less-tested features and improvements.
           prerelease: true
+
+      - name: Ensure release is published
+        run: gh release edit nightly --draft=false
```
Comment (raducristianpopa) [inline .github/workflows/nightly-build.yaml:57]: Is there going to be an issue if the release is not a draft?

Comment (sidvishnoi) [reply .github/workflows/nightly-build.yaml:57]: No, as I mentioned in PR description:
> This step won't fail if re-run on published release.

Comment (raducristianpopa) [reply .github/workflows/nightly-build.yaml:57]: My bad. Totally missed that ...

**Review by raducristianpopa — APPROVED** [review]

File: .github/workflows/nightly-build.yaml (hunk 1 above)
Comment (sidvishnoi) [reply .github/workflows/nightly-build.yaml:57]: No worries :)

**Pull request merged**